### PR TITLE
LF-3378: Repeat crop plan | Confirmation of repeated series

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- update French translation for RRule not to show "st", "nd", "rd" and "th" (based on [Calum's research](https://github.com/LiteFarmOrg/LiteFarm/pull/2769#discussion_r1268559780))
- break RepeatCropPlan util functions into smaller functions (get rule for text, occurrence), and create a new function to return text rule and occurrence rule
- create RepeatCropPlanConfirmation component and container
- update the type of some of RepeatCropPlan form's defaultValues (type 'number' will be 'string' once the value is updated)
- add `persistedFormData` to defaultValues so that user input persists after coming back from the confirmation view
- create stories (http://localhost:6006/?path=/docs/form-repeatcropplan-purerepeatcropplanconfirmation--docs)

**NOTE:**
- This PR can be fully tested with the code in https://github.com/LiteFarmOrg/LiteFarm/pull/2775.
- There is another [ticket](https://lite-farm.atlassian.net/browse/LF-3464) to complete UI/UX.

Jira link: https://lite-farm.atlassian.net/browse/LF-3378

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak
- [x] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/RepeatCropPlan/Confirmation.jsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  DAYS_OF_WEEK,
+  MONTH_REPEAT_ON,
+  FINISH,
+  AFTER_OCCURRENCES,
+  FINISH_ON_DATE,
+} from './constants';
+import styles from './styles.module.scss';
+
+/**
+ * Function that formats date to include day of week.
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/RepeatCropPlan/Confirmation.jsx:37]: This function looks generally useful. Consider adding to some top level util.

**Review by Duncan-Brain — APPROVED** [review]
@SayakaOno

Works seamlessly with the merged endpoint! Excellent job there!

Still seeing the scrollbar thing in storybook -- no one else?

I think this error was from some other PR but its still present here too:
Screenshot 2023-07-27 at 9 14 53 AM

File: packages/webapp/src/components/RepeatCropPlan/Confirmation.jsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      });
+    } else if (finish === 'after') {
+      finishingText = (
+        <Trans i18nKey={'REPEAT_PLAN.CONFIRMATION.AFTER_REPETITIONS'}>
+          after <b>{{ number: afterOccurrences }}</b> occurrences
+        </Trans>
+      );
+    }
+
+    const { text: willRepeatText, occurrences } = getTextAndOccurrences(
```
Comment (kathyavini) [inline packages/webapp/src/components/RepeatCropPlan/Confirmation.jsx:92]: Is the `willRepeatText` missing a reference to the RRule translations?

Screenshot 2023-07-27 at 8 46 18 AM

**Comment by SayakaOno** [comment]
@Duncan-Brain 
Thank you!!! I created a date util file.
We should be able to safely ignore the warning, I hope we don't have the limitation in the near future.
https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#limitations

File: packages/webapp/src/components/RepeatCropPlan/Confirmation.jsx (hunk 2 above)
Comment (SayakaOno) [reply packages/webapp/src/components/RepeatCropPlan/Confirmation.jsx]: Thank you so much for catching this!!!!!

**Review by kathyavini — APPROVED** [review]
It's great!!! ❤️

**Review by iperdomo — APPROVED** [review]

**Pull request merged**